tests/kani/Intrinsics/Count/ctlz.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+// Check that `ctlz` and `ctlz_nonzero` are supported and return the expected
+// results
+
+#![feature(core_intrinsics)]
+use std::intrinsics::{ctlz, ctlz_nonzero};
+
+// Define a function for counting like `ctlz` and assert that their results are
+// the same for any value
+macro_rules! test_ctlz {
+    ( $fn_name:ident, $ty:ty ) => {
+        fn $fn_name(x: $ty) -> $ty {
+            let mut count = 0;
+            let num_bits = <$ty>::BITS;
+            for i in 0..num_bits {
+                // Get value at index `i`
+                let bitmask = 1 << (num_bits - i - 1);
+                let bit = x & bitmask;
+                if bit == 0 {
+                    count += 1;
+                } else {
+                    break;
+                }
+            }
+            count
+        }
+        let var: $ty = kani::any();
+        // Check that the result is correct
+        assert!($fn_name(var) == ctlz(var));
+        // Check that the stable version returns the same value
+        assert!(ctlz(var) as u32 == var.leading_zeros());
+    };
+}
+
+// Assert that the results of `ctlz` and `ctlz_nonzero` are the same if we
+// exclude zero
+macro_rules! test_ctlz_nonzero {
+    ($ty:ty) => {
+        let var_nonzero: $ty = 8;
```

Comment:
Was this intended to be assigned to `kani::any()` instead of 8?

Reply:
Yes, thanks for catching it! Fixed 😄